src/Processor/Braintree/Repository/PaymentMethodRepository.php

```diff
@@ -29,7 +30,7 @@ public function add(PaymentMethod $paymentMethod): Token
             ->paymentMethod()
             ->create($request);
 
-        if (!$result->success) {
+        if ($result instanceof Error || !property_exists($result, 'paymentMethod')) {
```

Comment:
not all `instanceof Success` will have `paymentMethod` on them, since that's based off of what is being returned, I added it to a check here, so it passes below on line 37.